File: WordCountCBinder.java
//Needs java8
//javac WordCountCBinder.java
//java WordCountCBinder

import java.io.IOException;
import java.nio.file.FileSystems;
import java.nio.file.Files;
import java.nio.file.LinkOption;
import java.nio.file.Path;
import java.util.Collections;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

public class WordCountCBinder {
    public static void main(String[] args) throws IOException {
        String errorMsg = "Call program with: java WordCountCBinder -f <FileName> -ger|-eng";
        if(args.length < 3) {
            System.out.println(errorMsg);
            return;
        }

        String filename=null;
        String language=null;
        
        for(int i=0; i<args.length; i++) {
            String arg = args[i];
            if(arg.equals("-f") && (++i) < args.length) {
                filename = args[i];
                continue;
            }
        
            if(arg.equals("-ger"))
                language = "ger";
            else if (arg.equals("-eng"))
                language = "eng";
        }
        
        if(filename == null || language == null) {
            System.out.println(errorMsg);
            return;
        }
        
        Path path = FileSystems.getDefault().getPath(filename);

        if(Files.notExists(path, LinkOption.NOFOLLOW_LINKS)){
            System.out.println("Given input file must exist");
            return;
        }
        
        HashMap<String, Integer> map = new HashMap<String, Integer>();
        
        for (String line : Files.readAllLines(path)) {
            line = line.toLowerCase().replaceAll("'", "");
            
            if(language.equals("ger"))
                line = line.replaceAll("ö", "oe").replaceAll("ä", "ae").replaceAll("ü", "ue").replaceAll("ß", "ss");
            
            // http://stackoverflow.com/questions/1805518/replacing-all-non-alphanumeric-characters-with-empty-strings
            line = line.replaceAll("[^a-z0-9]", " ");
            
            for (String word : line.split(" ")) {
                if(word.length() == 0)
                    continue;
                
                int abundance = 0;
                if(map.containsKey(word))
                    abundance = map.get(word);
                
                map.put(word, abundance + 1);
            }    
        }
        
        // http://stackoverflow.com/a/23846961/2386941
        List<String> sortedWords = map.entrySet().stream()
                .sorted(Collections.reverseOrder(Map.Entry.comparingByValue()))
                .map(entry -> entry.getKey())
                .collect(Collectors.toList());
        
        String result = String.join(", ", sortedWords);

        System.out.println(result);
    }
}

File: WordCount-cBinder.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# #####
# python-version muss python 3 sein
# AUFRUF: > python WordCount-cBinder.py -f input.txt -ger
# entweder -ger oder -eng
# #####

import argparse
import os.path
from itertools import *


def handleArgs():
    """
    Argument handling, returning the input file and the language (either "ger" or "eng")
    :return: Tuple of input file and language string
    """
    parser = argparse.ArgumentParser(description='Counting words in an English or German text')
    parser.add_argument("-f", required=True, dest="filename", help="The filename of the input file")

    parser.add_argument("-ger", dest="language", action="store_const", const="ger",
                        help="The language of the input file is German")
    parser.add_argument("-eng", dest="language", action="store_const", const="eng",
                        help="The language of the input file is English")

    args = parser.parse_args()

    if not args.language:
        parser.error("No language defined, must be either -ger or -eng")

    if not os.path.isfile(args.filename):
        parser.error("Given file doesn't exist")

    return args.filename, args.language


def normalizeChar(character):
    character = character.lower()

    if character == "'":
        return ''

    if not character.isalnum():
        return ' '

    return character


def normalize(line, language):
    # http://stackoverflow.com/questions/12985456/replace-all-non-alphanumeric-characters-in-a-string
    line = ''.join([normalizeChar(s) for s in line]).strip()

    if language is "eng":
        return line

    line = line.replace('ö', 'oe').replace('ä', 'ae').replace('ü', 'ue').replace('ß', 'ss')
    return line


def main():
    fn, language = handleArgs()

    dictionary = dict()
    with open(fn, encoding='utf8') as fp:
        words = list(chain.from_iterable(normalize(line, language).split() for line in fp))
        for word in words:
            dictionary[word] = dictionary.get(word, 0) + 1

    # http://stackoverflow.com/questions/613183/sort-a-python-dictionary-by-value
    sortedByFrequency = sorted(dictionary, key=dictionary.get, reverse=True)
    print(', '.join(sortedByFrequency))


if __name__ == '__main__':
    main()
